src/codeu/chat/client/commandline/Chat.java

```diff
@@ -171,21 +188,31 @@ public void invoke(Scanner args) {
 
       // Find the first user with the given name and return a user context
       // for that user. If no user is found, the function will return null.
-      private UserContext findUser(String name) {
-        for (final UserContext user : context.allUsers()) {
-          if (user.user.name.equals(name)) {
-            return user;
-          }
-        }
-        return null;
-      }
+      
     });
 
+   
     // Now that the panel has all its commands registered, return the panel
     // so that it can be used.
     return panel;
   }
 
+  private UserContext findUser(String name, Context context) {
+    for (final UserContext user : context.allUsers()) {
+      if (user.user.name.equals(name)){
+        return user;
+      }
+    }
+    return null;
+  } 
+  private ConversationContext find(String title, UserContext user) {
+    for(final ConversationContext conversation : user.conversations()) {
+      if(title.equals(conversation.conversation.title)) {
```

Comment:
Done. I had to run a formatting script in all the code, so there are a lot of other changes regarding style as well. Sorry :/